### DISCUSSION TITLE
Linting is not working in my code

### DISCUSSION THREAD

**Discussion by karthiksathishjeemain** [Cyfrin/foundry-full-course-cu#1056] [Q&A]
I work on GitPod and earlier it use to show linting ( showing errors in the code by underlining it with red line automatically ), but after 3 days of starting the project, it stoped me showing any errors even though it consisted lot of errors. It raises error only when I compile it. Please help me

**Comment by theirrationalone** [comment]
Add extension `solidity` by nomic foundation, reload gitpod and check if linting is or not, if not then....
copy your whole codebase on a safe place.....
delete that gitpod project and try again by pasting it back...

**Comment by karthiksathishjeemain** [comment]
Still not solvable @theirrationalone . Please provide other solutions

**Reply by theirrationalone** [reply]
Shift to vscode local dev env if you could. Otherwise take a look into extension's settings... Read gitpod docs etc....

**Comment by robinpunn** [comment]
I'm using VSCode, and I was having the exact same issues. I realized it was happening after installing the OpenZeppelin contracts. I tried to follow as they said in their docs: ``Add @openzeppelin/contracts/=lib/openzeppelin-contracts/contracts/ in remappings.txt`` not just to foundry.toml. But that didn't help with the linting. 

I was using the Juan Blanco Solidity extension. While searching for how to fix the issue, I came across these:
https://github.com/juanfranblanco/vscode-solidity/issues/284
https://github.com/juanfranblanco/vscode-solidity/issues/274
Seems like the problem is related to there not being a ``node_modules`` folder in the foundry project. I couldn't solve the problem with any of the suggestions there.

I ended up removing the Juan Blanco extension and installing the Nomic Foundation extension. The linting works for me now.
